### ISSUE TITLE
Failed to send audio packet for decoding 3015, 3016 error

### ISSUE THREAD

**Issue by neerajgulati-airtel** [shaka-project/shaka-player#5692]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**If the problem is related to FairPlay, have you read the tutorial?**
NA

**What version of Shaka Player are you using?**
Shaka 4.4.2 and the issue is reproducible on nightly as well as 4.4.5

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
demo app

**If custom app, can you reproduce the issue using our demo app?**
yes

**What browser and OS are you using?**
macos and chrome

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
na

**What are the manifest and license server URIs?**
Sent on mail

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

**What did you do?**
The issue is intermittently reproducible for our users at normal playback speed and condtions. 

1) Play the url in shaka custom demo
2) Navigate to around 34 min and move right arrow key to take the video forward. Somewhere in 35 mins the playback gives error. Have seen errors like Max call stack size exceeded or append buffer failed or failed to send audio packet

When the video player downloads and tries to play the buffer it gives error

**What did you expect to happen?**
Playback to work fine

**What actually happened?**
The issue is intermittently reproducible for our users at normal playback speed and conditions. 

1) Play the url in shaka custom demo
2) Navigate to around 34 min and move right arrow key to take the video forward. Somewhere in 35 mins the playback gives error. Have seen errors like Max call stack size exceeded or append buffer failed or failed to send audio packet

This produces the same error which is intermittently occurring for our users under normal playback speed and conditions. Also I have reproduced it on nightly build 4.4.5 
<img width="1488" alt="Screenshot 2023-09-28 at 6 22 22 PM" src="https://github.com/shaka-project/shaka-player/assets/120081455/f2267538-16fb-4792-8acb-84b60f1be863">
<img width="1510" alt="Screenshot 2023-09-28 at 6 23 11 PM" src="https://github.com/shaka-project/shaka-player/assets/120081455/70bffc1c-560b-4af2-8416-89a35a41c0f1">
as well.

**Comment by avelad** [comment]
@neerajgulati-airtel The stream has been expired... Can you resend a new stream? Thanks!

**Comment by neerajgulati-airtel** [comment]
Hi,

PFB

https://hunstream.hungama.com/c/5/09d/866/1335042/1335042_,750,100,400,1000,1600,3000,6500,.mp4.m3u8?2ihPXWq3z1f-mJxnFQmaCDFLaZB1MvCPONmaFhq6-ROqs9SnVX1XgpAjGKiU8fJg2h6ry-pGgQdr3SQPyAApQqrX3ZyylGe_6YZHjw6sKkhoMwLO9iVRsHcUJw

Regards,
Neeraj Gulati

**Comment by neerajgulati-airtel** [comment]
Have sent the url again in mail

**Comment by avelad** [comment]
@neerajgulati-airtel does the error occur in all qualities or only in some?

**Comment by avelad** [comment]
I can reproduce the bug in 4.4.2 but not in the main branch. We recently (yesterday) fixed several issues that could have fixed the problem.

**Comment by neerajgulati-airtel** [comment]
Hi I am able to reproduce  it on nightly v4.4.2-main-21-g025502a70 deployed at https://nightly-dot-shaka-player-demo.appspot.com. I hope yesterday changes are present on it. I am able to reproduce it on auto quality. 
<img width="1507" alt="Screenshot 2023-09-29 at 1 50 56 PM" src="https://github.com/shaka-project/shaka-player/assets/120081455/85a0589d-fbf4-453a-b7e0-0cf824550bfe">

**Comment by avelad** [comment]
Can you tell me what version of Chrome and macOS you are using? It would also be good to know if you use an Mx or Intel processor. Thanks!

**Comment by neerajgulati-airtel** [comment]
MacOS version: 13.5.2 (22G91)
Chip: Apple M1 Pro
Chrome Version: 117.0.5938.132

**Comment by neerajgulati-airtel** [comment]
@avelad  Let me know when you pick this for debugging, I will share the urls again

**Comment by avelad** [comment]
For the next two weeks I won't be able to work on this, sorry, I have other priorities.

**Comment by avelad** [comment]
I've fixed some transmuxer issues on the main branch, can you check if it's still happening? thank you!

**Comment by neerajgulati-airtel** [comment]
@avelad, I just checked and is getting the same error on shaka custom demo

<img width="1510" alt="image" src="https://github.com/shaka-project/shaka-player/assets/120081455/ff27b3c5-7a79-4695-9771-458339dda709">
Have shared the manifest url in mail

**Comment by avelad** [comment]
I've fixed some transmuxer issues on the **main branch**, can you check if it's still happening? thank you!

**Comment by neerajgulati-airtel** [comment]
No issue is still there
<img width="1510" alt="image" src="https://github.com/shaka-project/shaka-player/assets/120081455/417d2435-932d-45b0-95d0-1a80f18f780d">

Have shared latest manifest url in mail

**Comment by avelad** [comment]
Can you test if the problem still occurs with version 4.7? Thank you!

**Comment by avelad** [comment]
@neerajgulati-airtel Can you send the url to shaka-player-maintainers@googlegroups.com? I'll try to give it a second review this week.

**Comment by neerajgulati-airtel** [comment]
Just sent

**Comment by avelad** [comment]
@neerajgulati-airtel A question, using the quality selector, does it occur in all the qualities or only in a specific one?

**Comment by neerajgulati-airtel** [comment]
Occuring irrespective of quality

**Comment by avelad** [comment]
I tried to reproduce the error on mac intel, mac m2 pro and m3, and I cannot reproduce the error. Also @neerajgulati-airtel mentions that the same problem ocurrs with hls.js. I tried with hls.js in mac intel,  mac m2 pro and m3 and the problem doesn't occur. 
I don't have a mac m1 pro to test it, so I cannot help more... @neerajgulati-airtel are you interested on try to find a solution and send a PR to fix it? Thanks!
